drivers/digital-io/max22196/README.rst

```diff
@@ -0,0 +1,147 @@
+MAX22196 no-OS driver
```

Comment:
please link in this commit also the readme file to the drivers ToC (after #1955 gets merged, to avoid possible conflicts)